File: Models/Entities/Student.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;
using System.Diagnostics;

namespace DiplomaThesisDigitalization.Models.Entities
{
    public class Student
    {
        [Key, ForeignKey("User")]
        public int Id { get; set; }
        public int ECTS { get; set; }
        public string DegreeLevel { get; set; }
        [JsonIgnore]
        public User User { get; set; }
        public int? DiplomaThesisId { get; set; }

        public DiplomaThesis DiplomaThesis { get; set; }

        public int FieldId { get; set; }  // Specializimi ~~ Field
        public Field Field { get; set; }
        public int DepartmentId { get; set; }
        public Department Department { get; set; }
    }
}

File: my-ts-react-app/src/app/models/Student.ts
import { Department } from "./Department";
import { DiplomaThesis } from "./DiplomaThesis";
import { Field } from "./Field";
import { User } from "./User";

export interface Student {
  id: number;
  ects: number;
  degreeLevel: string;
  name: string;
  surname: string;
  email: string;
  diplomaThesisId: number | null;
  diplomaThesis: DiplomaThesis | null;
  fieldId: number;
  field: Field;
  departmentId: number;
  department: Department;
}
